sdk/tables/azure-data-tables/test/ut/sas_test.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "sas_test.hpp"
+
+#include <chrono>
+#include <sstream>
+#include <string>
+#include <thread>
+
+using namespace Azure::Data::Tables::Sas;
+// cspell: words rwdxylacupitf raud bqft
+namespace Azure { namespace Data { namespace Test {
+  TEST(SasTest, TableSasBuilderTestAllSet)
+  {
+    TablesSasBuilder sasBuilder;
```

Comment:
I'm going to ask an obnoxious question.

The `Builder` pattern is a fairly well known pattern where the builder class has a set of methods which (for instance) do things like:

```c++

TablesSasBuilder sasBuilder;
sasBuilder
  .SetPermissions(TablesSasPermissions::All)
  .SetStartsOn(...)
  .SetExpiresOn(...)
  .SetIdentifier("myIdentifier")
  .SetIPRange("iprange")

 :
 :
```

Typically classes which implement the "Builder" pattern have a suffix of "Builder".

When I saw `TablesSasBuilder`, I assumed that it implemented the Builder pattern, so I was a bit surprised to see that it didn't.

Is there a reason to call this a builder given that it doesn't follow the builder pattern? Possibly something like SasBuilderFactory (which would imply it followed the factory pattern which is a different pattern altogether).

Anyway, I'm not asking for a change here, but it IS something to think about. If (for instance), the other languages have a similar shape to this shape, I'm ok with this. But I'd like to understand how this type looks in all the existing languages.

Reply:
let m think about it, do we have examples of this pattern ? i suspect w might have something

Reply:
The builder pattern is here: https://www.geeksforgeeks.org/builder-pattern-c-design-patterns/ 

Interestingly, this version of the builder pattern and a couple of others does not have the "return *this;" part of the pattern which allows you to chain settings on the builder (other examples I've found do).

The `AccountSasBuilder` is similar to the pattern as described, except it uses mutable fields on the builder rather than methods.

In general, we don't use the Builder pattern in the C++ (we do in other languages like Java and Rust because they're common patterns in those languages). 

IMHO the Storage team mis-used the name "Builder", I would have called them on the use just like I'm calling out the use here.

Humorously, the .Net SDK equivalent uses the "Builder" name, but the Java equivalent does not (Humorously because Java embraces the Builder pattern :))

Because this naming is prevalent in the Storage SDKs as well as .Net (the other P0 languages have different conventions), I can tolerate the name in this case, but its use is questionable elsewhere.

PS: We might want to consider creating an `AccountSasCredential` in `Azure::Core` to express the various SAS credential options for all the services which support SAS credentials (similar to what is done in the other languages)

Reply:
that would be a good step , as this is a common functionality with "flavors" for services, i wonder if we could break it/ templetize it such the the permissions /services enums could be fed in based on the consumer service and generalize the generate method to accomodate the various signature strings and sas parameters

Reply:
filed issue https://github.com/Azure/azure-sdk-for-cpp/issues/5436 to track this and not forget about it

Reply:
I agree the name implies a pattern that we end up not respecting. What would be your suggestion , i could spend some time implementing it or rename the objects , again sugestion for the name ?